### DISCUSSION TITLE
Is there a way to specify/override the default io-manager (`io_manager` resource key) for all the assets of a specific job, while preserving all the resources already defined in the main `Definitions`?

### DISCUSSION THREAD

**Discussion by dagsir** [dagster-io/dagster#18196] [Q&A] [answered]
Is there a way to specify/override the default io-manager (`io_manager` resource key) for all the assets of a specific job, while preserving all the resources already defined in the main `Definitions`?

---

*The question was originally asked in [Dagster Slack](https://dagster.slack.com/archives/C01U954MEER/p1700588301977749?thread_ts=1700588301.977749&cid=C01U954MEER).*

**Comment by benpankow** [comment, marked as answer]
Right now, this is not yet possible. Each asset's `io_manager_key` would have to be set to a custom key, which can then be specified in your `Definitions` file.
